projects/netgrif-components-core/src/lib/navigation/utility/navigation-item-task-utility-methods.ts

```diff
@@ -92,3 +97,37 @@ export function extractFilterFromFilterField(filterField: FilterField): Filter {
     }
     return SimpleFilter.fromQuery({query: filterField.value}, filterField.filterMetadata.filterType);
 }
+
+/**
+ * Extracts the selected search type from enumeration field of the navigation item task data.
+ * @returns a {@link SearchMode} containing {@link SearchMode.ADVANCED} or {@link SearchMode.FULLTEXT} or {@link undefined}
+ * if unexpected value is found
+ * */
+export function extractSearchTypeFromData(dataSection: Array<DataGroup>, typeFieldId: string): SearchMode {
+    const typeField = getFieldFromDataGroups(dataSection, typeFieldId);
+    if (typeField === undefined) {
+        throw new Error('Navigation entry search type field could not be resolved');
+    }
+
+    switch (typeField.value) {
+        case 'fulltext':
+            return SearchMode.FULLTEXT;
+        case 'fulltext_advanced':
```

Comment:
is this the only place, these constants are used? If not, they should be extracted

Reply:
this is the only place of usage